### PR TITLE
Create index.d.ts

### PR DESCRIPTION
**Pull request by BjMrq**
Hello, working with the package on one off my project I realized there were no types (and the one that is @types/joi-password-complexity is no well designed) 
Any review / comment is welcome!

### CODE REVIEW

**Comment by kamronbatman** [comment]
Thanks!

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+declare module 'joi-password-complexity'{
+  import { ValidationError } from '@hapi/joi';
```
Comment (kamronbatman) [inline index.d.ts:2]: We need to support the new joi also.
`const joi = softRequire('joi') || softRequire('@hapi/joi');`
How can we go about that?

Comment (BjMrq) [reply index.d.ts:2]: Hummmm I Typescript does not seem to be able to type it's way through with getJoi. 
I could re-type the ErrorValidation so we don't need imports

Comment (BjMrq) [reply index.d.ts:2]: Like so 

```javascript
  interface ValidationErrorItem {
    message: string;
    type: string;
    path: string[];
    options?: ValidationOptions;
    context?: Context;
  }

  interface ValidationError extends Error {
    details: ValidationErrorItem[];
    annotate(): string;
    _object: any;
  }
```

Comment (kamronbatman) [reply index.d.ts:2]: That should work

**Pull request merged**